thoth/storages/graph/postgres.py

```diff
@@ -3630,6 +3630,52 @@ def get_kebechet_github_installations_active_managers(self, slug: str) -> list:
                 return active_managers
             return []
 
+    def get_kebechet_github_installation_info_for_software_environment(
+        self,
+        *,
+        python_version: Optional[str] = None,
+        image_name: Optional[str] = None,
+        image_sha: Optional[str] = None,
+        os_name: Optional[str] = None,
+        os_version: Optional[str] = None,
+        thoth_s2i_image_name: Optional[str] = None,
+        thoth_s2i_image_version: Optional[str] = None,
+        env_image_name: Optional[str] = None,
+        env_image_tag: Optional[str] = None,
+        cuda_version: Optional[str] = None,
+        environment_type: Optional[str] = None,
+    ):
+        with self._session_scope() as session, session.begin(subtransactions=True):
+            query = (
+                session.query(KebechetGithubAppInstallations)
+                .filter(KebechetGithubAppInstallations.is_active.is_(True))
+                .join(ExternalPythonSoftwareStack)
+            )
+
+            if python_version:
+                query = query.filter(ExternalSoftwareEnvironment.python_version == python_version)
+            if image_name:
+                query = query.filter(ExternalSoftwareEnvironment.image_name == image_name)
+            if image_sha:
+                query = query.filter(ExternalSoftwareEnvironment.image_sha == image_sha)
+            if os_name:
+                query = query.filter(ExternalSoftwareEnvironment.os_name == os_name)
```

Comment:
@fridex regarding https://github.com/thoth-station/storages/issues/2196#issuecomment-860744514, do we want to adjust the logic to have also only rhel and not ubi to be consistent? in that case here there should `map_os_name` method to have rhel if someone (bot or human) uses ubi8, which is what is present in .thoth.yaml and what purge job consider as source of truth in EcosystemSolver Table.

Reply:
+1 on the remapping

Reply:
So we need to adjust sync logic in SoftwareEnvironment and ExternalSoftwareEnvironment Tables.. actually all tables that have `os_name` :D

Reply:
Sounds good 👍🏻